lib/Qubit.class.php

```diff
@@ -214,4 +214,33 @@ public static function likeISO8601Date($date)
       return false;
     }
   }
+
+  /**
+   * Check if the app is in read only mode looking at the
+   * env. variable "ATOM_READ_ONLY" and the setting
+   * "read_only" in app.yml, giving priority to the env. variable
+   *
+   * @return boolean
+   */
+  public static function isReadOnly()
+  {
+    // The env. variable is set to 'true', '1', 'on' or 'yes'
+    if (filter_var(getenv('ATOM_READ_ONLY'), FILTER_VALIDATE_BOOLEAN))
+    {
+      return true;
+    }
+
+    // The env. variable doesn't exist and the setting is set to 'true'
+    if (false === getenv('ATOM_READ_ONLY')
+      && sfConfig::get('app_read_only', false))
+    {
+      return true;
+    }
+
+    // Otherwise:
+    // - The env. variable and the setting don't exist
+    // - The env. variable is set to anything different than 'true', '1', 'on' or 'yes'
+    // - The env. varibale doesn't exist and the setting is set to anything different than 'true'
+    return false;
+  }
```

Comment:
I think that it would be more elegant to have QubitSettingsFilter looking at $_ENV, so we can keep interfacing with sfConfig from our code, i.e. I'd rather keep interfacing with sfConfig::get('app_whatever') that introducing new methods to retrieve individual settings like your suggested Qubit::isReadOnly().

QubitSettingsFilter avoids repetitive access to the database with cache. Once that's done (#L38), you can iterate over a list of known environment variables. I think that it makes sense to define an internal hashmap where the key is the environment variable and the value is its type, e.g. "ATOM_READ_ONLY" => "boolean", "ATOM_FOOBAR" => "string", etc...

Thoughts?